File: app/src/main/java/im/tox/antox/callbacks/AntoxOnFileDataCallback.java
package im.tox.antox.callbacks;

import android.content.Context;
import android.util.Log;

import im.tox.antox.utils.AntoxFriend;
import im.tox.jtoxcore.callbacks.OnFileDataCallback;
import im.tox.jtoxcore.callbacks.OnFileSendRequestCallback;

public class AntoxOnFileDataCallback implements OnFileDataCallback<AntoxFriend> {

    private static final String TAG = "OnFileDataCallback";
    private Context ctx;

    public AntoxOnFileDataCallback(Context ctx) { this.ctx = ctx; };

    public void execute(AntoxFriend friend, int filenumber, byte[] data) {
        Log.d(TAG, "execute");
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnFileDataCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import android.util.Log
import im.tox.antox.callbacks.AntoxOnFileDataCallback._
import im.tox.antox.tox.ToxSingleton
import im.tox.antox.utils.AntoxFriend
//remove if not needed

object AntoxOnFileDataCallback {

  private val TAG = "OnFileDataCallback"
}

class AntoxOnFileDataCallback(private var ctx: Context) {

  def execute(friend: AntoxFriend, filenumber: Int, data: Array[Byte]) {
    Log.d(TAG, "execute")
    ToxSingleton.receiveFileData(friend.getAddress, filenumber, data, ctx)
  }
}
